src/Api.php

```diff
@@ -304,6 +302,60 @@ protected function query($uri, array $data = [], $method = 'GET', array $headers
   }
 
   /**
+   * Performs http queries to Fastly API server (VCL related).
+   *
+   * @param string $uri
+   *   The uri to use for the request, appended to the host.
+   * @param array $data
+   *   (optional) Data to send with the request.
+   * @param string $method
+   *   (optional) The method to use for the request, defaults to GET.
+   * @param array $headers
+   *   (optional) An array of headers to send with the request.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   *   Response.
+   *
+   * @throws \GuzzleHttp\Exception\RequestException
+   *   RequestException.
+   */
+  protected function VQuery($uri, array $data = [], $method = 'GET', array $headers = []) {
```

Comment:
Yes, I believe this needs some additional adjusting. Note that I'm fairly new to Drupal 8 (I use to work with it very long time ago so I'm still catching up).
In this module I was actually trying to port vcl functionality from WP (https://github.com/fastly/WordPress-Plugin/blob/master/src/classes/vcl-handler.php) and this ended up little bit more tricky than I initially thought. Being new to fastly, this looked like good idea at the time.
Anyway, I'm little more into Drupal and Fastly now and I'll hurry up with changes.
Thanks for very helpful comments.